chainer/links/normalization/batch_normalization.py

```diff
@@ -66,22 +66,28 @@ def __init__(self, size, decay=0.9, eps=2e-5, dtype=numpy.float32,
                  use_gamma=True, use_beta=True,
                  initial_gamma=None, initial_beta=None, use_cudnn=True):
         super(BatchNormalization, self).__init__()
+        # [cuDNN] dtype of prameters like gamma, beta, etc. must be float32,
+        # even when dtype of input/output tensors are float16.
+        dtype_param = dtype
+        if use_cudnn and dtype_param == numpy.float16:
+            dtype_param = numpy.float32
```

Comment:
Oh, i think the current type-check code of batch-normalization function forbids this type of argument. See https://github.com/pfnet/chainer/pull/2299/files#diff-148edea535bb806260c487ca31869a91R64